components/ShareDog.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import styled from 'styled-components/native'
+import { Share, View, Button } from 'react-native'
+
+
+
+export const ShareDog = () => {
+  const onShare = async () => {
```

Comment:
Congratulations about `async await` usage :)
For further explanation of this topic you can check out https://medium.com/javascript-in-plain-english/async-await-javascript-5038668ec6eb